### ISSUE TITLE
Make /daily-note/create?silent=false open note even if it already existed

### ISSUE THREAD

**Issue by johtso** [czottmann/obsidian-actions-uri#47]
Currently the behaviour is inconsistent and only opens the note if it was newly created. This means that if you want to ensure the note exists and open it, you have to call 2 action URIs (create and then open-current) to make sure it happens.

Does a change in behaviour here seem sensible?

**Comment by czottmann** [comment]
@johtso Thanks for the report!

The current behavior is correct because if a current DN exists, trying to create a note without allowing it to be overwritten results in a fail state. It shouldn't be "failed, _but_...".

Adding a toggle to `/daily-note/open-current` would make it deviate from its sister route `/daily-note/open-most-recent` which I don't like.

**That being said,** how about setting up a new route, `/daily-note/create-or-open-current`? One optional argument `content` that would be ignored if the DN already exists.

**Comment by geoelectric** [comment]
I didn’t file the original, but as I just wandered into the GitHub site looking for this functionality, sounds great to me. 

I think the one (not particularly compelling) argument towards giving the `daily-note/create` an “open if exists” option is that the equivalent official route,`/new`, takes `&append` to do exactly that. But then they have to document the conflict with `&overwrite`, etc, and I agree it becomes less cohesive.

Overall, I prefer your approach.

**Comment by czottmann** [comment]
@geoelectric Whoa, wall of text. 😅 Thanks for your input! At the moment, I'm knee-deep in an adjacent project with higher priority, so I just skimmed it — but I want 7 need to find some time to really think this through in order to form an actual opinion here.

Give me a few days, please.

**Comment by geoelectric** [comment]
Yeah, sorry about that. The proposal sort of grew out as I thought it through. I did try to divide it up a little for comprehension, at least. By the time I was done maybe it should’ve been markdown headers…

TL;DR: one route, consistent with your idea with a couple tweaks, allows ubiquitous logging from anywhere and opens up interapp connection with a logic free one-liner.

**Comment by czottmann** [comment]
@geoelectric I had some time to think it through, apologies for the delay. **It's a good idea.** Here's a sketch for a route `/daily-note/create-or-append` (I've omitted writing down the description for its `/note/create-or-append` counterpart).

For consistency, I'd keep the existing `/daily-note/create` and `/note/create` routes but adjust their signature — i.e., I'd add an `ignore-template` parameter and therefore the default behaviour.

Let me know what you think. 

---

Finds or creates the current daily and appends a piece of text. If the note has to be created first, Obsidian will use the template that is specified in its settings, unless the  `ignore-template` parameter is set.

An existing current daily note **will not** be overwritten **unless** the related parameter is explicitly set.

The route `/daily-note/create-or-append` would look as follows:
  - `ignore-template`: optional boolean, defaults to `false`
  - `content`: optional string, text to append
  - `overwrite`: optional boolean
  - `silent`: optional boolean

### Scenarios:

- Note exists, `overwrite` is not set
  - `ignore-template` has no function
  - `content` contains text
    - The content is appended to the note.
  - `content` is empty
    - Nothing happens
- Note exists, `overwrite` is set
  - Existing note is ignored, so see "Note doesn't exist" below
- Note doesn't exist
  - `overwrite` has no function
  - `ignore-template` is set
    - `content` contains text
      - A blank note is created, the content is appended to the note.
    - `content` is empty
      - A blank note is created
  - `ignore-template` is not set
    - `content` contains text
      - A note is created based on a template per Obsidian's settings, the content is appended to the note.
    - `content` is empty
      - A note is created based on a template per Obsidian's settings

**Comment by johtso** [comment]
@czottmann this sounds perfect!

**Comment by jancbeck** [comment]
Not sure if this is helps but I found this issue while googling for my use case where I want to append a text to the bottom of the daily note. If a daily note exists for today then just use that or else create a new one from the template (which has a "Notes:" header at the bottom of it). 

I'm using this setup in Apple Shortcuts

<img width="627" alt="Screenshot 2023-03-18 at 18 15 30" src="https://user-images.githubusercontent.com/865311/226122548-5170ade4-1bfe-4789-98f1-2a343eeec63d.png">

Note how I'm calling /dailynote/create first with no content and then call /dailynote/append with the x-callback and x-success. This ensures that a daily note exists to append to. This works on Mac but not on mobile so I'll have to investigate further. But maybe it helps some struggling with the same issue.

**Comment by czottmann** [comment]
@jancbeck At first I was like "Why would you call `obsidian://` as a callback‽" and then I understood. This is a clever idea for chaining!

Obligatory plug: if you're open to trade juggling XCU calls in Shortcuts with way more convenience, check out my app [Actions for Obsidian](https://obsidian.actions.work). ;)

**Comment by czottmann** [comment]
I've released [v0.18](https://github.com/czottmann/obsidian-actions-uri/releases/tag/0.18.0) which overhauls the "create or overwrite or error" mechanic by adding a new `if-exists` parameter which supersedes the `overwrite` parameter. I think it should alleviate a lot of the pain points.

> The[/daily-note/create](https://czottmann.github.io/obsidian-actions-uri/routes/daily-note/#daily-notecreate) route has a new optional if-exists parameter to specify a strategy for dealing with an existing current daily note. It overrides the default behavior (returning an error) and can be set to skip or overwrite. if-exists=skip will pretend the existing note was just created and return it. if-exists=overwrite will trash the existing note and create a new daily note from scratch.

**Comment by jancbeck** [comment]
Thanks for adding this! 

For others who might need this, I'm posting my [shortcut](https://support.apple.com/en-gb/guide/shortcuts/welcome/ios):

<img width="633" alt="Screenshot 2023-06-02 at 10 29 16" src="https://github.com/czottmann/obsidian-actions-uri/assets/865311/69fa653a-5931-4c7b-8047-23db500aaec4">

I put this as an icon on my home screen for when I want to quickly save a thought to my journal.
I also have an automation that asks to run this every evening before winddown.
